src/routes/api/bookings/deleteBooking.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response } from 'express';
+import BookingModel from '../../../models/booking.model';
+import TimeBlockModel from '../../../models/timeBlock.model';
+
+const deleteBookingRoute = async (req: Request, res: Response): Promise<void> => {
+    const id = req.params.id;
+
+    // Retrieves the booking of the given id
+    const booking = await BookingModel.findOne({ _id: id }).catch((error) => {
+        res.status(500).json({ error });
+        return;
+    });
+
+    if (!booking) {
+        res.status(404);
+        return;
+    }
+
+    // Returns an error if the user deleting the booking is not the same as the user who made the booking
+    if (req.userEmail != booking.booker) {
+        res.status(403).json({ error: 'user is not the booker of this booking so cannot delete it' });
+    }
+
+    // Retrieves the time block that corresponds to the booking
+    const bookingTimeBlock = await TimeBlockModel.findOne({ _id: booking.timeBlock })
+        .populate('bookings')
+        .catch((error) => {
```

Comment:
Also can let Express handle it here, the return only works on the promise so the rest of the code would still be ran